scripts/release.sh

```diff
@@ -314,10 +329,59 @@ main() {
     exit 1
   fi
 
-  # TODO: signing process
-  log_warning ""
-  log_warning "WARNING: The release has not been signed and published to github. This must be done manually."
-  log_warning ""
+  if [ "${DRY_RUN}" == "true" ] || [ "${NO_GH_RELEASE}" == 1 ]; then
+    log_warning ""
+    log_warning "WARNING: Skipping creating GitHub release, --no-gh-release is set."
+    log_warning "WARNING: If not running on DRY_MODE, please do the GitHub release manually."
+    log_warning ""
+  else
+    local gh_repo
+    local release_notes_temp_file
+    local release_url
+    local gh_release_args=()
+
+    case "${BRANCH}" in
+      "main")
+        gh_release_args=(--prerelease)
+        ;;
+      "release-3.5")
+        gh_release_args=(--latest)
+        ;;
+    esac
+
+    if [ "${REPOSITORY}" = "$(pwd)" ]; then
+      gh_repo=$(git remote get-url origin)
+    else
+      gh_repo="${REPOSITORY}"
+    fi
+
+    gh_repo=$(echo "${gh_repo}" | sed 's/^[^@]\+@//' | sed 's/https\?:\/\///' | sed 's/\.git$//' | tr ':' '/')
+    log_callout "Creating GitHub release for ${RELEASE_VERSION} on ${gh_repo}"
+
+    release_notes_temp_file=$(mktemp)
+
+    # Disable sellcheck SC2016, the single quoted syntax for sed is intentional.
+    # shellcheck disable=SC2016
+    sed 's/${RELEASE_VERSION}/'"${RELEASE_VERSION}"'/' ./scripts/release_notes.tpl.txt > "${release_notes_temp_file}"
+    maybe_run gh release create "${RELEASE_VERSION}" \
+        --repo "${gh_repo}" \
+        --draft \
+        --title "${RELEASE_VERSION}" \
+        --notes-file "${release_notes_temp_file}" \
+        "${gh_release_args[@]}" \
+        ./release/SHA256SUMS
+
+    # Upload files one by one, as gh doesn't support passing globs as input.
+    maybe_run find ./release '(' -name '*.tar.gz' -o -name '*.zip' ')' -exec \
+      gh --repo "${gh_repo}" release upload "${RELEASE_VERSION}" {} \;
```

Comment:
We also need to upload `SHA256SUMS`

Reply:
This is uploaded in line https://github.com/etcd-io/etcd/pull/18649/files#diff-1bba00e5aca52286a667c09eff92ba2ca8e3ca77e0d31b0599cc829ab0173389R372.

However, your comment made me realize that these steps I wrote are not stateless, and if we re-run the script after a failure, it won't be able to recover/continue drafting the GitHub release. I'll address that issue in this PR.

Reply:
Done